cli/cmd/network.go

```diff
@@ -0,0 +1,171 @@
+package cmd
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/spf13/cobra"
+
+	"github.com/oasisprotocol/oasis-sdk/cli/cmd/common"
+	"github.com/oasisprotocol/oasis-sdk/cli/config"
+	"github.com/oasisprotocol/oasis-sdk/cli/table"
+)
+
+var (
+	networkCmd = &cobra.Command{
+		Use:   "network",
+		Short: "Manage network endpoints",
+	}
+
+	networkListCmd = &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "List configured networks",
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := config.Global()
+			table := table.New()
+			table.SetHeader([]string{"Name", "Chain Context", "RPC"})
+
+			var output [][]string
+			for name, net := range cfg.Networks.All {
+				output = append(output, []string{
+					name,
+					net.ChainContext,
+					net.RPC,
+				})
+			}
+
+			// Sort output by name.
+			sort.Slice(output, func(i, j int) bool {
+				return output[i][0] < output[j][0]
+			})
+
+			table.AppendBulk(output)
+			table.Render()
+		},
+	}
+
+	networkAddCmd = &cobra.Command{
+		Use:   "add <name> <chain-context> <rpc-endpoint>",
+		Short: "Add a new network",
+		Args:  cobra.ExactArgs(3),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := config.Global()
+			name, chainContext, rpc := args[0], args[1], args[2]
+
+			// Ask user for some additional parameters.
+			questions := []*survey.Question{
+				{
+					Name:   "description",
+					Prompt: &survey.Input{Message: "Description:"},
+				},
+				{
+					Name:   "symbol",
+					Prompt: &survey.Input{Message: "Denomination symbol:"},
+				},
+				{
+					Name: "decimals",
+					Prompt: &survey.Input{
+						Message: "Denomination decimal places:",
+						Default: "9",
+					},
+					Validate: survey.Required,
+				},
+			}
+			answers := struct {
+				Description string
+				Symbol      string
+				Decimals    uint8
+			}{}
+			err := survey.Ask(questions, &answers)
+			cobra.CheckErr(err)
+
+			err = cfg.Networks.Add(name, &config.Network{
+				Description:  answers.Description,
+				ChainContext: chainContext,
+				RPC:          rpc,
+				Denomination: config.DenominationInfo{
+					Symbol:   answers.Symbol,
+					Decimals: answers.Decimals,
+				},
+			})
+			cobra.CheckErr(err)
+
+			err = cfg.Save()
+			cobra.CheckErr(err)
+		},
+	}
+
+	networkRmCmd = &cobra.Command{
+		Use:     "rm <name>",
```

Comment:
This doesn't need to be resolved in this PR, but could the shell completion be configured to list all the networks' names when one presses <kbd>Tab</kbd> after typing:
```
oasis network rm
```

Reply:
Yeah this should be done in a separate PR, feel free to file an issue once this is merged.